src/data/finset/basic.lean

```diff
@@ -1705,6 +1716,18 @@ end list
 
 namespace finset
 
+lemma exists_list_nodup_eq [decidable_eq α] (s : finset α) :
+  ∃ (l : list α), l.nodup ∧ l.to_finset = s :=
```

Comment:
Note that for `multiset` we have `multiset.to_list` which essentially serves the same purpose as this lemma:

https://github.com/leanprover-community/mathlib/blob/6ee8f0e27f9bf253cd16235367eaecaae86dc5ab/src/data/multiset/basic.lean#L258-L275

Do you think it makes sense to add this to finset? Or do we already have a `finset.to_list` under another name?

Reply:
So you propose that instead of 
```
obtain ⟨l, hl, rfl⟩ := s.exists_list_nodup_eq,
rw cycle_factors_finset_eq_list_to_finset hl,
```
I do the following?
```
let l : list α := s.to_list,
have hl : l.nodup := s.to_list_nodup,
have hl' : s = l.to_finset := by simpa [←s.to_list_iff_to_finset]
rw [hl', cycle_factors_finset_eq_list_to_finset hl]
```

Reply:
Only if you think that's better. At a glance, it looks like it's probably worse. Would you mind linking to `cycle_factors_finset_eq_list_to_finset `?

Reply:
https://github.com/leanprover-community/mathlib/compare/pechersky/cycle-factors-finset#diff-821c5a206b688540db806a7f3a471120b21a9ba73781d2d34bdc8e6f3aabd957R600-R612

Reply:
I think having a bundled exists lemma is more convenient that a separate noncomputable def. If someone wants to really have a separate def, they can `classical.some` out of it.

Reply:
I like this existential statement.